app/components/inputs/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,143 @@
+import { ImageListType, ImageType } from "react-images-uploading";
+import { useState } from "react";
+import axios from "axios";
+import { toast } from "react-hot-toast";
+import { PreviewImageWithUrl } from "@/app/types";
+
+interface Props {
+  value: PreviewImageWithUrl[];
+  onChange: (value: PreviewImageWithUrl[]) => void;
+}
+
+export const useUploadImage = ({ value: images, onChange }: Props) => {
+  const [previewImages, setPreviewImages] = useState(
+    images.map((image) => {
+      const { url, ...rest } = image;
+      return { ...rest, dataURL: url as string } as ImageType;
+    })
+  );
+
+  const [isUploading, setIsUploading] = useState(false);
+
+  const uploadImage = async (imageData: string) => {
```

Comment:
It's better to put `uploadImage` and `onImageFileChange` outside of the `useUploadImage` function, to make the `useUploadImage` logic more readable.

Reply:
Sure I can take it out